src/sys_process_lifecycle.c

```diff
@@ -0,0 +1,97 @@
+/*
+Copyright (C) 2015 The University of Notre Dame
+This software is distributed under the GNU General Public License.
+See the file LICENSE for details.
+*/
+
+#include "console.h"
+#include "process.h"
+#include "iso.h"
+#include "memorylayout.h" // PROCESS_ENTRY_POINT
+#include "sys_process_lifecycle.h"
+
+uint32_t sys_exit(uint32_t code) {
+    process_exit(code);
+    return 0;
+}
+
+uint32_t sys_yield() {
+    process_yield();
+    return 0;
+}
+
+uint32_t sys_run(char *process_path, struct process_permissions *child_permissions) {
+    if (child_permissions == 0) {
+        return -1;
+    }
+
+    // Load process data
+    struct iso_dir *root_dir = iso_dopen("/", 3);
```

Comment:
It seems weird that we require process to be loaded from ATA devices 3. Can we not pass the device number in, or otherwise indicate which device is desired in the file path?